test/src/test/java/org/corfudb/infrastructure/log/BoundedMapTest.java

```diff
@@ -0,0 +1,131 @@
+package org.corfudb.infrastructure.log;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class BoundedMapTest {
+
+    @Test
+    public void testIndexSet() {
+        BoundedMap index = new BoundedMap(0, 10_000);
+        assertThat(index.contains(10L)).isFalse();
+        assertThat(index.set(10, 1000L)).isTrue();
+        assertThat(index.set(10, 1000L)).isFalse();
+        assertThat(index.get(10)).isEqualTo(1000L);
+        assertThat(index.capacity()).isEqualTo(10_000);
+    }
+
+    @Test
+    public void testSetOffsetAndLength() {
+        BoundedMap index = new BoundedMap(0, 10_000);
+        long address = 100;
+        long offset = 1000;
+        int length = 2048;
+        assertThat(index.set(address, offset, length)).isTrue();
+        long packed = index.get(address);
+        assertThat(BoundedMap.unpackLength(packed)).isEqualTo(length);
+        assertThat(BoundedMap.unpackOffset(packed)).isEqualTo(offset);
+
+        long address2 = 2000;
+
+        assertThat(index.set(address2, 0x0000000fffffffffL, 0x0fffffff));
```

Comment:
In this case, I noticed that the BoundedMap cannot detect that this address has already been set since the packed result is also equal to NOT_SET. It doesn't seem like a problem since the metadata's serialized size will not be that large, but it wasn't clear if this is expected behaviour. In either case, consider validating the above with contains().

Reply:
Yea, that's a good point. For a general map, where the whole bit range can be used, but in our case there are invalid ranges where the offset cannot be true AND the buffer size cannot be 0.

Essentially, a value that is equal to NOT_SET cannot not be used to set an index, I made it more explicit by checking the value and reserving Long.MIN_VALUE.